acp/ACP.md

```diff
@@ -0,0 +1,419 @@
+# Introduction
```

Comment:
thought: I found this document amazingly comprehensive, to the point at which I was laughing at it being in a PR. IMO it is very useful to us and users, and could/should probably become the ACP user guide?

I do worry a little as to how maintainable it is, and if it over does it a little bit for the target audience (particularly in the Introduction), however if you are happy with it I'm happy as-is :)

I also have a minor preference for backticks over `**` in a bunch of places (mostly in the `DefraDB's Access Control System` section), but don't go out of your way to change them I think.

Reply:
I had ACP user guide in mind as well while writing it. Glad you liked it.

- [ ] Maybe change to backticks over `**`

Reply:
UPDATE: Changed all to backticks